lambda/handler_test.go

```diff
@@ -15,6 +15,15 @@ import (
 
 func TestInvalidHandlers(t *testing.T) {
 
+	type valuer interface {
+		Value(key interface{}) interface{}
+	}
+
+	type customContext interface {
+		context.Context
+		MyCustomMethod()
```

Comment:
You mention in the PR description that you wanted to be able to use this kind of interface with your handlers but if all aws-lambda-go is passing in is a `valuer` (ie, a context.Context) wouldn't this fail at runtime because `context.Context` does not satisfy `customContext`?

Looking at the test it must not fail at runtime but I don't understand exactly how.

Reply:
I want to use interfaces like `valuer`, but I don't need to use `customContext`.

> wouldn't this fail at runtime because context.Context does not satisfy customContext?

Yes, actually it panics with "reflect: Call using *context.valueCtx as type main.customContext" in aws/aws-lambda-go.
Users can’t understand it is a bug of aws/aws-lambda-go or wrong usage of aws/aws-lambda-go.
so, aws/aws-lambda-go should return an error with more friendly message rather than panicking.